lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -910,4 +911,55 @@ public function getUpdateSchemaSql(array $classes, $saveMode = false)
 
         return $schemaDiff->toSql($this->platform);
     }
+
+    /**
+     * Creates the schema from the database, ensuring tables from the target schema are whitelisted for comparison.
+     *
+     * @param Schema $toSchema The target schema.
+     *
+     * @return Schema The schema from the database.
+     */
+    private function createSchemaForComparison(Schema $toSchema) : Schema
+    {
+        $sm = $this->em->getConnection()->getSchemaManager();
+
+        // check and memorize if we are using dbal >= 2.9
+        static $hasFilterMethod;
+        $hasFilterMethod  = $hasFilterMethod ?? method_exists(Configuration::class, 'getSchemaAssetsFilter');
+        $config           = $this->em->getConnection()->getConfiguration();
+        $filter           = null;
+        $filterExpression = null;
+
+        if ($hasFilterMethod) {
+            // check and memorize if we are using dbal >= 3.0
+            static $hasFilterExpressionMethod;
+            $hasFilterExpressionMethod = $hasFilterExpressionMethod ?? method_exists(Configuration::class, 'getFilterSchemaAssetsExpression');
+
+            // backup schema assets filter and filter-expression
+            $filterExpression = $hasFilterExpressionMethod ? $config->getFilterSchemaAssetsExpression() : null;
+            $filter           = $config->getSchemaAssetsFilter();
+
+            if ($filter !== null) {
+                // whitelist assets we already know about in $toSchema, use the existing filter otherwise
+                $config->setSchemaAssetsFilter(static function ($asset) use ($filter, $toSchema) : bool {
+                    $assetName = $asset instanceof AbstractAsset ? $asset->getName() : $asset;
+
+                    return $toSchema->hasTable($assetName) || $toSchema->hasSequence($assetName) || $filter($asset);
+                });
+            }
+        }
+
+        try {
+            return $sm->createSchema();
+        } finally {
+            if ($hasFilterMethod && $filter !== null) {
+                // restore schema assets filter and filter-expression
+                if ($filterExpression !== null) {
+                    $config->setFilterSchemaAssetsExpression($filterExpression);
+                } else {
+                    $config->setSchemaAssetsFilter($filter);
+                }
+            }
```

Comment:
We released the last version of the `2.6.x` series to be able to better focus on the `2.7.x` series, which means that we're making `2.7.x` the target of this PR.

I was actually hoping to release this on `2.7.0` but the complexity of this method makes me hesitate. The DBAL 2.9+ check isn't needed since ORM 2.7 requires DBAL 2.9+, which means we can clean this up.